tests/test-sss.C

```diff
@@ -0,0 +1,295 @@
+    FFLAS::fflas_delete(Id);
+
+    return ok;
+}
+
+/** \brief test equality between applying productSSSxTS or dense matrix */
+template<class Field>
+bool test_application_compatibility (const Field & F, size_t n, size_t t, size_t s,
+                                     const typename Field::Element alpha,
+                                     typename Field::ConstElement_ptr P, size_t ldp,
+                                     typename Field::ConstElement_ptr Q, size_t ldq,
+                                     typename Field::ConstElement_ptr R, size_t ldr,
+                                     typename Field::ConstElement_ptr U, size_t ldu,
+                                     typename Field::ConstElement_ptr V, size_t ldv,
+                                     typename Field::ConstElement_ptr W, size_t ldw,
+                                     typename Field::ConstElement_ptr D, size_t ldd,
+                                     typename Field::ConstElement_ptr B, size_t ldb,
+                                     const typename Field::Element beta,
+                                     typename Field::ConstElement_ptr C, size_t ldc)
+{
+    typename  Field::Element_ptr A = fflas_new (F, n, n);
+    SSSToDense (F, n, s, P, ldp, Q, ldq, R, ldr, U, ldu, V, ldv, W, ldw,
+                D, ldd, A, n);
+  
+    typename  Field::Element_ptr dense = fflas_new (F, n, t);
+    typename  Field::Element_ptr qs = fflas_new (F, n, t);
+    fassign (F, n, t, C, ldc, dense, t);
+    fassign (F, n, t, C, ldc, qs, t);
+
+    fgemm (F, FflasNoTrans, FflasNoTrans, n, t, n, alpha, A, n, B, ldb, beta, dense, t);
+    productSSSxTS (F, n, t, s, alpha, P, ldp, Q, ldq, R, ldr, U, ldu, V, ldv, W, ldw, D, ldd,
+                   B, ldb, beta, qs, t);
+
+    bool ok = fequal (F, n, t, dense, t, qs, t);
+
+    FFLAS::fflas_delete(A);
+    FFLAS::fflas_delete(dense);
+    FFLAS::fflas_delete(qs);
+
+    if ( !ok )
+        std::cout << "Different results for dense and qs application "<<std::endl;
+    else
+        std::cout << "Same results for dense and qs application "<<std::endl;
+    return ok;
+}
+
+template<class Field>
+bool run_with_field(Givaro::Integer q, uint64_t b, size_t n, size_t s, size_t t, size_t iters, uint64_t seed)
+{
+    std::cout << "Test new type" << std::endl;
+    bool ok = true ;
+    typedef typename Field::Element_ptr Element_ptr ;
+    while (ok && iters)
+    {
+        std::cout << "Iterations left: " << iters << std::endl;
+            /* New field 
+             * chooseField returns a pointer, F needs to be passed by its value */
+        Field* F= chooseField<Field>(q,b,seed);
+        if (F==nullptr)
+            return true;
+            /* Initiate random number generator */
+        typename Field::RandIter G(*F,seed++);
+
+            /* Generate generators (which can be cropped a bit) */
+        Element_ptr D = fflas_new (*F, n, s);
+        Element_ptr P = fflas_new (*F, n, s);     // Could be n - s 
```

Comment:
when changing the dimensions to the suggested ones, the testsuite fails

Reply:
Suggestions were wrong. Generators are now minimized to the correct sizes and the tests pass.